File: src/lib/models/DTO/Domain/BlogPostDto.cs
using System;
using System.Collections.Generic;

namespace DTO.Domain
{
    public class BlogPostDto : BaseDomainDto
    {
        public string Title{ get; set; }
        public string Content{ get; set; }
        public long ImgId{ get; set; }
        public ICollection<FocusBlogPostMappingDto> Foci{ get; set; }

        public UserDto CreationUser{ get; set; }
        public ImgDto Img{ get; set; }

        new public bool Validate(bool throwException = false)
        {
            var invalid = base.Validate()
                && string.IsNullOrEmpty(this.Title)
                || string.IsNullOrEmpty(this.Content)
                || CreationUserId <= 0
                || (this.Img == null && this.ImgId <= 0);

            if(invalid && throwException)
            {
                throw new Exception($"Invalid {nameof(BlogPostDto)} structure");
            }

            return invalid;
        }
    }
}

File: src/ui/asq-ui/src/app/_models/domain/blog-post-dto.ts
import { BaseDomainDto } from './base-domain-dto';
import { UserDto } from './user-dto';
import { ImgDto } from './img-dto';
import { FocusBlogPostMappingDto } from './focus-blog-post-mapping-dto';

export class BlogPostDto extends BaseDomainDto{
    public title: string = '';
    public content: string = '';
    public creationUserId: number = 0;
    public imgId: number = 0;
    public foci: FocusBlogPostMappingDto[] = null;

    public creationUser: UserDto = null;
    public img: ImgDto = null;

    constructor(){
        super();
    }
}
